test/whatsapp/api/client_test.rb

```diff
@@ -62,6 +62,45 @@ def test_invalid_api_version
         end
       end
 
+      def test_logs_http_requests_into_the_logger_without_response_body
+        logger_io = StringIO.new
+        client = Client.new('test_token', ApiConfiguration::DEFAULT_API_VERSION, Logger.new(logger_io))
+
+        stub_test_request(:get)
+        client.send_request(endpoint: 'test', http_method: 'get')
+
+        logged_string = logger_io.string
+        middleware_handlers = faraday_middlewares(client)
+
+        assert_includes(middleware_handlers, Faraday::Response::Logger)
+        assert_match_logger_output!(logged_string)
+        assert_match('INFO -- response: ', logged_string)
+      end
+
+
+      def test_logs_http_requests_into_the_logger_with_response_body
+        logger_io = StringIO.new
+        client = Client.new(
+          'test_token',
+          ApiConfiguration::DEFAULT_API_VERSION,
+          Logger.new(logger_io),
+          { bodies: true }
+        )
+
+        stub_test_request(:get)
+        client.send_request(endpoint: 'test', http_method: 'get')
+
+        logged_string = logger_io.string
+        assert_match_logger_output!(logged_string)
+        assert_match('INFO -- response: {"success":true}', logged_string)
+      end
+
+      def test_should_not_log_when_logger_not_configured
+        # Validates whether the logger was configured into faraday or not
+        middleware_handlers = faraday_middlewares(@client)
+        refute_includes(middleware_handlers, Faraday::Response::Logger)
```

Comment:
nit: you can use `StringIO.new` to assert there wasn't any log.

Reply:
hmmm yup I considered this option, but I thought that way the test case won't ever be failed regardless of the code changes :sweat_smile: 
What do you think about it?

Reply:
not required, just an alternative.